lib/OpenQA/JobTemplates.pm

```diff
@@ -21,12 +21,46 @@ use Exporter 'import';
 use Carp;
 use Try::Tiny;
 use JSON::Validator;
+use YAML::PP 0.020;
 
 our $VERSION   = '0.0.1';
 our @EXPORT_OK = qw(
-  &validate_data
+  &validate_data &load_yaml &dump_yaml
 );
 
+my $yp;
+
+sub _init_yaml_processor {
+    $yp ||= YAML::PP->new(
```

Comment:
Wouldn't a state variable work better here? Otherwise it should be `$YP` as a package scope variable.

Reply:
Any reason not to initialize the object when the package is loaded? It seems fairly cheap.

Reply:
I initialized it at the beginning now and named it `$YP`